plugin_oar.py

```diff
@@ -49,90 +20,140 @@ def __init__(self, config):
-
-
-def _read_oarnodes_s_yaml(fn_s, write_method):  # todo: fix write_method not being used
-    assert os.path.isfile(fn_s)
-    anonymize = anonymize_func()
-    logging.debug('File %s exists: %s' % (fn_s, os.path.isfile(fn_s)))
-    try:
-        assert os.stat(fn_s).st_size != 0
-    except AssertionError:
-        logging.critical('File %s is empty!! Exiting...\n' % fn_s)
-        raise
-    data = yaml.safe_load(fn_s, DEF_INDENT=4)
-    if options.ANONYMIZE:
-        nodes_resids = dict([(anonymize(node, 'wns'), resid_state.items()) for node, resid_state in data.items()])
-    else:
-        nodes_resids = dict([(node, resid_state.items()) for node, resid_state in data.items()])
-    # import wdb; wdb.set_trace()
-    return nodes_resids
-
-
-def _read_oarnodes_y_yaml(fn_y):
-    with open('oarnodes_y', mode='r') as fin:
-        data = yaml.load(fin)
-    resids_jobs = dict([(resid, info.get('jobs', None)) for resid, info in data.items()])
-    return resids_jobs
-
-
-def _read_oarnodes_y_textyaml(fn):
-    oar_nodes = {}
-    logging.debug("Before opening %s" % fn)
-    with open(fn, mode='r') as fin:
-        logging.debug("File state %s" % fin)
-        fin.readline()  # '---'
-        line = fin.readline().strip()  # first res_id
-        while line:
-            oar_node, line = _read_oar_node_y_textyaml(fin, line)
-            oar_nodes.update(oar_node)
-
-        resids_jobs = dict([(resid, info.get('jobs', None)) for resid, info in oar_nodes.items()])
-    return resids_jobs
-    # return oar_nodes
-
-
-def _read_oar_node_y_textyaml(fin, line):
-    _oarnode = dict()
-
-    res_id = line.strip(': ')
-    _oarnode[int(res_id)] = dict()
-
-    line = fin.readline().strip()
-    while line and not line[0].isdigit():
-        key, value = line.strip().split(': ')
-        _oarnode[int(res_id)][key] = value
-        line = fin.readline().strip()
+                qstat_values = self._process_line(re_search, line, re_match_positions)
+                all_values.append(qstat_values)
+
+        logging.debug('File state after OarStatMaker.serialise_qstat: %(fin)s' % {"fin": fin})
+        self.dump_all(all_values, out_file, write_method)
+
+
+class OARBatchSystem(object):
+    def __init__(self, in_out_filenames, config):
+        self.oarnodes_s_file = in_out_filenames.get('oarnodes_s_file')
+        self.oarnodes_y_file = in_out_filenames.get('oarnodes_y_file')
+        self.oarstat_file_out = in_out_filenames.get('oarstat_file_out')
+        self.oarstat_file = in_out_filenames.get('oarstat_file')
+
+        self.config = config
+        self.oar_stat_maker = OarStatMaker(self.config)
+
+    def convert_inputs(self):
+        return self._serialise_qstat()
+
+    def get_worker_nodes(self):
+        return self._get_worker_nodes(self.oarnodes_s_file, self.oarnodes_y_file)
+
+    def get_jobs_info(self):
+        return common_module.get_jobs_info(self.oarstat_file_out)
+
+    @staticmethod
+    def get_queues_info():
```

Comment:
Not sure where `get_queues_info()` is used, but it's static in one class and non-static on the others which might imply broken interface.